core/src/main/java/org/apache/calcite/rex/RexCall.java

```diff
@@ -288,6 +288,11 @@ private Pair<SqlOperator, List<RexNode>> getNormalized() {
     }
     Pair<SqlOperator, List<RexNode>> x = getNormalized();
     RexCall rexCall = (RexCall) o;
+    // RANDOM() f unction calls are never equal
+    if (this.getOperator().getKind() == SqlKind.RANDOM
+        || rexCall.getOperator().getKind()  == SqlKind.RANDOM) {
+      return false;
+    }
```

Comment:
I assume this check is for if it's literally the same object in java:

```
if (this == o)  {
  return true;
}
```

HOWEVER, for our purposes, do we want to make the object not equal itself if it's random? IE, should we reorder these clauses?